pkg/apis/sources/v1alpha1/cloudeventingress_types.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// CloudEventIngressSpec defines the desired state of CloudEventIngress
+type CloudEventIngressSpec struct {
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Sink is the URI
+	Sink *corev1.ObjectReference `json:"sink,omitempty"`
+	// ServiceAccountName is the name of the ServiceAccount to use to run this
+	// source.
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+}
+
+// CloudEventIngressStatus defines the observed state of CloudEventIngress
+type CloudEventIngressStatus struct {
```

Comment:
Please see https://github.com/knative/eventing-sources/pull/258 for migrating to the ducktype for embedded status.

Reply:
Updated. Please review the new version.